compiler+runtime/test/jank/cpp/try/pass-catch-cpp-std-bad-cast.jank

```diff
@@ -0,0 +1,17 @@
+;; Test catching std::bad_cast
+(cpp/raw "
+#include <typeinfo>
+void throw_bad_cast() {
+  throw std::bad_cast();
+}")
+
+(let [caught (atom false)]
+  (try
+    (cpp/throw_bad_cast)
+    (catch cpp/std.runtime_error e
```

Comment:
We're missing a test which catches an exception by its base verifies no slicing happened. 

Details: https://wiki.sei.cmu.edu/confluence/display/cplusplus/ERR61-CPP.+Catch+exceptions+by+lvalue+reference